### PR TITLE
feat: use sf-plugin-core base class SfCommand (#384)

### PR DESCRIPTION
**Pull request by RasPhilCo**
### What does this PR do?
Subclasses all commands to use SfCommand base class

### What issues does this PR fix or reference?

W-9880969

### CODE REVIEW

**Comment by RasPhilCo** [comment]
@RodEsp Hola! I'm not sure if you can help us here, but when we moved to SfCommand base class in this PR, the disable json flag property stopped working. We are adding json flags to commands in other PR's, so I'm concerned about merge this to master with not-yet-implemented json flags. cc @saracope

**Comment by saracope** [comment]
what we're seeing:
Screen Shot 2022-04-26 at 6 00 12 PM

**Comment by RodEsp** [comment]
Sorry for the delayed response @RasPhilCo, I was at TDX last week. Will be taking a look at this today though.

**Review by RodEsp — COMMENTED** [review]
@RasPhilCo that's because the property is named `enableJsonFlag` in `SfCommand`.

If you make this change you should be good to go 👍🏼

File: src/lib/base.ts [hunk 1]
```diff
@@ -5,28 +5,28 @@
...
 import APIClient, { herokuClientApiUrl } from './api-client';
 import herokuVariant from './heroku-variant';
 import { SfdcAccount } from './sfdc-types';
 
-export default abstract class Command extends Base {
+export default abstract class Command extends SfCommand<any> {
   protected static TOKEN_BEARER_KEY = 'functions-bearer';
   protected static TOKEN_REFRESH_KEY = 'functions-refresh';
   // We want to implement `--json` on a per-command basis, so we disable the global json flag here
   static disableJsonFlag = true;
```
[the remark this answers is not in the document]
Comment (saracope) [reply src/lib/base.ts]: Thanks!

**Pull request merged**